h/static/scripts/tests/controllers/search-bar-controller-test.js

```diff
@@ -138,6 +193,97 @@ describe('SearchBarController', function () {
         });
     });
 
+    describe('it allows group value suggestions', function () {
+
+      const getLozengeValues = function (){
+        return Array.from(testEl.querySelectorAll('.lozenge')).map((el)=>{
+          return el.querySelector('.lozenge__content').textContent;
+        });
+      };
+
+      beforeEach(function(){
+        // we need to setup the env vars before invoking controller
+        teardown();
+        addGroupSuggestions();
+        setup();
+
+        sinon.stub(testEl.querySelector('form'), 'submit');
+      });
+
+      it('shows group suggestions', function(done){
+        syn
+          .click(input)
+          .type('group:', () => {
+            assert.isTrue(dropdown.classList.contains('is-open'));
+
+            let titles = getItemTitles();
+
+            assert.lengthOf(titles, 5, 'we should be enforcing the 5 item max');
+          })
+          .type('[backspace][backspace][backspace][backspace][backspace][backspace]', () => {
+            assert.deepEqual(getItemTitles(), [ 'user:', 'tag:', 'url:', 'group:' ], 'tags go away as facet is removed');
+            done();
+          });
+      });
+
+      it('orders groups by indexOf score', function(done){
```

Comment:
What is "indexOf score"? Ideally tests should be phrased in terms of the behavior that the end user sees so that during any refactoring, it is obvious what the consequences of a failing test are from a user's POV.